### PR TITLE
HDDS-11632. Publish images to GitHub container repo

### PR DESCRIPTION
**Pull request by adoroszlai**
## What changes were proposed in this pull request?

Workflow to build and publish multi-platform `ozone` Docker image to GitHub Container Registry and Docker Hub.

- Pull request: image is built, but not published
- Push: image is built and published by commit SHA.
- Push to branch `ozone-X.Y`: the image is also published with Docker tag `X.Y`.  This matches current Docker Hub automated build rules.
- Similarly to `ozone-docker-runner`, push to Docker Hub is optional, only performed if credentials are available.

This PR is targeted at `latest`, which is like `main` for this repo.  After merge, we'll need to cherry-pick it to version-specific branches where we would like the workflow to be used.  We'll also need to ask Infra to disable the automated build at Docker Hub.

https://issues.apache.org/jira/browse/HDDS-11632

## How was this patch tested?

Push to branches:

- `HDDS-11632` ([workflow](https://github.com/adoroszlai/ozone-docker/actions/runs/11730671311), [image](https://github.com/adoroszlai/ozone-docker/pkgs/container/ozone/302215411?tag=2721d2e5503dfa7b797ff9bfc8e6f5c21fa487fc)).
- `ozone-test` ([workflow](https://github.com/adoroszlai/ozone-docker/actions/runs/11730787985), [image](https://github.com/adoroszlai/ozone-docker/pkgs/container/ozone/302215411?tag=test)).
- `ozone-test-with-dh` after adding Docker Hub credentials to the repo ([workflow](https://github.com/adoroszlai/ozone-docker/actions/runs/11730972980), image at [ghcr.io](https://github.com/adoroszlai/ozone-docker/pkgs/container/ozone/302215411?tag=test-with-dh) and [hub.docker.com](https://hub.docker.com/layers/adoroszlai/ozone/test-with-dh/images/sha256-e5d20b95e06eac5e6d5b6116e26d5689bed9c68d650167aac52fa122b5a048cd?context=repo))

(Note that the same commit was pushed to all branches, so tags `test` and `test-with-dh` refer to the same image.)

### CODE REVIEW

**Comment by adoroszlai** [comment]
@errose28 @smengcl please take a look

Even if approved, I'll wait for response on INFRA-26258 before merging this.

File: .github/workflows/build-and-tag.yaml
Comment (smengcl) [inline .github/workflows/build-and-tag.yaml]:
```suggestion
          echo "REGISTRIES=${{ env.REGISTRIES }} docker.io" >> $GITHUB_ENV
```

**Review by smengcl — APPROVED** [review]
Thanks @adoroszlai . just one nit. feel free to merge this when infra issue is resolved.

**Comment by adoroszlai** [comment]
Thanks @smengcl for the review.  Infra issue is fixed.

**Pull request merged**
